library/acl.inc

```diff
@@ -864,6 +864,34 @@
     return $s;
   }
 
+
+  // Returns array of all ACOs
+  function gen_aco_array() {
+    global $phpgacl_location;
```

Comment:
This is very similar to function gen_aco_html_options. It would be better to put most of this logic into a single function and avoid code duplication.